### ISSUE TITLE
nullspace returns (basis, nullity), but documentation claims (nullity, basis)

### ISSUE THREAD

**Issue by cossio** [wbhart/Nemo.jl#59]
In the documentation it is claimed that `nullspace` returns a pair (d, V), where `d` is the dimension of the kernel and `V` a basis. In fact, `nullspace` returns the pair in the opposite order, `(V, d)`.

**Comment by wbhart** [comment]
Thanks. This is a documentation bug. We will fix this in the next version.

Also feel free to submit a PR when you find things like this. You can also
make pull requests for small fixes like this directly on GitHub these days
too, which is probably the easiest thing.

Bill.

**Comment by cossio** [comment]
@wbhart Also note that `nullspace(x::Nemo.fmpz_mat)` and `nullspace{T<:Nemo.FieldElem}(M::Nemo.MatElem{T<:Nemo.FieldElem})` return the pair in opposite orders!

**Comment by cossio** [comment]
@wbhart Maybe it is not a documentation bug. Maybe it's just `nullspace(x::Nemo.fmpz_mat)` that needs to return in the correct order?

**Comment by wbhart** [comment]
Yes, the fmpz_mat one looks incorrect. The person who implemented it
probably didn't know which order the generic code returned them, and I
didn't check when I merged it.

Thanks for the extra info!
